include/crypto/evp.h

```diff
@@ -191,6 +191,32 @@ struct evp_kdf_st {
     OSSL_OP_kdf_set_ctx_params_fn *set_ctx_params;
 };
 
+struct evp_rand_st {
+    OSSL_PROVIDER *prov;
+    int name_id;
+    CRYPTO_REF_COUNT refcnt;
+    CRYPTO_RWLOCK *lock;
+
+    const OSSL_DISPATCH *dispatch;
+    OSSL_OP_rand_newctx_fn *newctx;
+    OSSL_OP_rand_freectx_fn *freectx;
+    OSSL_OP_rand_instantiate_fn *instantiate;
+    OSSL_OP_rand_uninstantiate_fn *uninstantiate;
+    OSSL_OP_rand_generate_fn *generate;
+    OSSL_OP_rand_reseed_fn *reseed;
+    OSSL_OP_rand_nonce_fn *nonce;
+    OSSL_OP_rand_set_callbacks_fn *set_callbacks;
+    OSSL_OP_rand_enable_locking_fn *enable_prov_locking;
+    OSSL_OP_rand_lock_fn *prov_lock;
+    OSSL_OP_rand_unlock_fn *prov_unlock;
+    OSSL_OP_rand_gettable_params_fn *gettable_params;
+    OSSL_OP_rand_gettable_ctx_params_fn *gettable_ctx_params;
+    OSSL_OP_rand_settable_ctx_params_fn *settable_ctx_params;
+    OSSL_OP_rand_get_params_fn *get_params;
+    OSSL_OP_rand_get_ctx_params_fn *get_ctx_params;
+    OSSL_OP_rand_set_ctx_params_fn *set_ctx_params;
+};
+
```

Comment:
Please place this in `crypto/evp/evp_local.h`, there's no need for wider exposure.  (I really don't think that `evp_mac_st` and `evp_kf_st` need this kind of exposure either)

`evp_md_st`, `evp_cipher_st` and a few others is a different matter, they need to be exposed more widely as long as we have constant EVP_MD, EVP_CIPHER and so on arrays.